pkg/prometheus/resource_selector.go

```diff
@@ -186,22 +185,20 @@ func selectObjects[T configurationResource](
 			logger.Warn("skipping object", "error", err.Error(), "object", namespaceAndName)
 			rs.eventRecorder.Eventf(obj, v1.EventTypeWarning, operator.InvalidConfigurationEvent, "%q was rejected due to invalid configuration: %v", namespaceAndName, err)
 		}
-		res = append(res, struct {
+		res[namespaceAndName] = struct {
 			resource T
-			key      string
 			err      error
 			reason   string
 		}{
 			resource: o,
-			key:      namespaceAndName,
 			err:      err,
 			reason:   reason,
-		})
+		}
 	}
 
 	keys := []string{}
-	for _, r := range res {
-		keys = append(keys, r.key)
+	for k := range res {
```

Comment:
don't we have a bug here? keys will contain the names of all resources including the invalid ones.